examples/nemo/scripts/utils.py

```diff
@@ -300,6 +300,17 @@ def insert_explicit_loop_parallelism(
                 " 'nlay_i' or 'nlay_s'.")
             continue
 
+        # If we have a region_trans this is already tested
+        if not region_directive_trans:
+            not_pure = [call.routine.name for call in loop.walk(Call)
```

Comment:
I'd be happier if we could put this check inside the appropriate transformation(s?) rather than in this utils.py. Is that possible?

Reply:
Done, I didn't do it the first time because I wasn't sure if it would impact lfric, but in retrospective builtin and kern are not calls so it has no impact. Even if we change them to being calls, the pure attribute should be a requirement for them.

Reply:
That's a good idea (marking kernels as PURE). We should raise it with @TeranIvy and @christophermaynard once they're back.